burst/src/K8sApi/K8sHPAMetricsService.cs

```diff
@@ -48,31 +49,52 @@ public K8sHPAMetrics GetK8SHPAMetrics(string ns, string deployment)
                 return null;
             }
 
-            K8sHPAMetrics hpaMetrics = new ();
-
             // If _hpaList is not null, we don't have any HPA
             if (hpaList.Items.Count == 0)
             {
                 logger.LogWarning("No HPA object found in any namespace");
             }
             else
             {
+                K8sHPAMetrics hpaMetrics = new ();
                 foreach (V2beta2HorizontalPodAutoscaler hpa in hpaList.Items)
                 {
                     if (hpa.Namespace().Equals(ns) && hpa.Name().Equals(deployment))
                     {
-                        // Get the Target CPU load
-                        hpaMetrics.TargetCPULoad = GetTargetCpuLoad(hpa);
+                        try
+                        {
+                            // Get the Target CPU load
+                            hpaMetrics.MaxLoad = GetMaxLoad(hpa);
+
+                            // Get the current CPU load
+                            hpaMetrics.CurrentLoad = GetCurrentLoad(hpa);
+                            hpaMetrics.TargetLoad = (int?)Math.Floor(hpaMetrics.MaxLoad.GetValueOrDefault() * TargetPercent);
+                        }
+                        catch (Exception ex)
+                        {
+                            logger.LogWarning(ex.Message);
+
+                            // Should have all values available
+                            // Otherwise return null
+                            return null;
+                        }
+
+                        // If calculated target load is zero (it can be since we are flooring MaxLoad)
+                        if (hpaMetrics.TargetLoad == 0)
+                        {
+                            hpaMetrics.TargetLoad = hpaMetrics.MaxLoad;
+                        }
 
-                        // Get the current CPU load
-                        hpaMetrics.CurrentCPULoad = GetCurrentCpuLoad(hpa);
                         return hpaMetrics;
                     }
                 }
             }
 
-            // At the very least return empty metrics
-            return hpaMetrics;
+            // TODO: What happens if we can't find hpa in this NS?
```

Comment:
@jofultz What happens if we can't find the HPA? Say, the deployment or the namespace doesn't exist. Or we don't have RBAC permission in that specific NS/Deployment.

Do we still return `null` from this function?
In that case controller will return 204 no-content.

Should we consider 404 for no-hpa found?